jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -512,33 +512,27 @@ private void checkAccess(Class<?> targetClass, String name, int memberModifiers,
 		 * @throws IllegalAccessException If the {@link Class} is not accessible from {@link #accessClass}. 
 		 */
 		private void checkClassAccess(Class<?> targetClass) throws IllegalAccessException {
-			/*[IF Sidecar19-SE]*/
-			checkClassModuleVisibility(accessMode, accessClass.getModule(), targetClass);
-			/*[ENDIF]*/
-			if (NO_ACCESS != accessMode) {
-				/* target class should always be accessible to the lookup class when they are the same class */
-				if (accessClass == targetClass) {
-					return;
-				}
-				
-				int modifiers = targetClass.getModifiers();
-				
-				/* A protected class (must be a member class) is compiled to a public class as
+			int targetModifiers = targetClass.getModifiers();
+			boolean accessible = false;
+			if (Modifier.isPublic(targetModifiers) || Modifier.isProtected(targetModifiers)) {
+				/*[IF Sidecar19-SE]*/
```

Comment:
Doesn't this refactoring also mean that a Lookup with `NO_ACCESS` can still access public types?

Reply:
Yes.  There is a lot of redundant checking for `NO_ACCESS` in the existing code.  I am 
looking at refactoring it.

Reply:
Let's get the fix for this in before looking at a larger refactoring

Reply:
Okay.